benchmarks/playfair_cipher/oop_c#/Program.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System;
+using System.Text;
+using System.Drawing;
+using System.Text.RegularExpressions;
+using System.IO;
+
+namespace oop_c_
+{
+    class Program
+    {
+        static string TEST_STRING = File.ReadAllText("benchmarks/playfair_cipher/lines.txt");
+        static void Main(string[] args)
+        {
+            PlayFairCipher p = new PlayFairCipher("this is a great keyword");
+            string encrypt = p.Encrypt(TEST_STRING);
+            System.Console.WriteLine(encrypt.Length);
+            string decrypt = p.Decrypt(encrypt);
+            System.Console.WriteLine(decrypt.Length);
+        }
+    }
+
+    public class Table
+    {
+        private char[,] charTable;
+        private Point[] positions;
+        private Point currentPos;
+        public Table()
+        {
+            charTable = new char[5, 5];
+            positions = new Point[26];
+            currentPos = new Point();
+        }
+
+        public void AddNext(char c)
+        {
+            charTable[currentPos.X, currentPos.Y] = c;
+            positions[c - 'A'] = currentPos;
+            if (currentPos.Y == 4 && currentPos.X != 4)
+            {
+                currentPos.X++;
+                currentPos.Y = 0;
+            }
+            else if (currentPos.X == 5)
+                throw new IndexOutOfRangeException();
+            else
+                currentPos.Y++;
+        }
+
+        public bool ContainsChar(char c) => positions[c - 'A'] != Point.Empty || charTable[0,0] == c;
+
+        public Point GetPositionFromChar(char c) => positions[c - 'A'];
+        public char GetCharFromPosition(int x, int y) => charTable[x,y];
+    }
+
+    public class PlayFairCipher
+    {
+        private Table charTable { get; set; }
+        private Point[] positions { get; set; }
+
+        public PlayFairCipher(string key)
+        {
+            string alphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+            charTable = new Table();
+            createTable(preprocessText(key + alphabet));
+        }
+
+        public string Encrypt(string text)
+        {
+            StringBuilder sb = new StringBuilder(preprocessText(text));
+
+            for (int i = 0; i < sb.Length; i += 2)
+            {
+                // If length is odd add X
+                if (i == sb.Length - 1)
+                    sb.Append(sb.Length % 2 == 1 ? "X" : "");
+
+                // If two adjacent characters are the same insert X inbetween
+                else if (sb[i] == sb[i+1])
+                    sb.Insert(i + 1, 'X');
+            }

```

Comment:
```suggestion
            for (int i = 0; i < sb.Length; i += 2)
            {
                // If two adjacent characters are the same insert X inbetween
                else if (sb[i] == sb[i+1])
                    sb.Insert(i + 1, 'X');
            }
            // If length is odd add X     
            sb.Append(sb.Length % 2 == 1 ? "X" : "");

```
I stedet for at tjekke for hver to karakterer, så bare tjekke efter X'ere er indsat?